sequence-core/src/main/java/com/power4j/kit/seq/persistent/provider/PostgreSqlSynchronizer.java

```diff
@@ -0,0 +1,223 @@
+	@Override
+	protected Optional<Long> selectSeqValue(Connection connection, String name, String partition) throws SQLException {
+		final String sql = POSTGRESQL_SELECT_VALUE.replace("$TABLE_NAME", tableName);
+		log.debug(sql);
+		try (PreparedStatement statement = connection.prepareStatement(sql)) {
+			statement.setString(1, name);
+			statement.setString(2, partition);
+			log.debug(String.format("param: [%s] [%s]", name, partition));
+			updateCount.incrementAndGet();
+			try (ResultSet resultSet = statement.executeQuery()) {
+				if (resultSet.next()) {
+					if (resultSet.getObject(1) == null) {
+						throw new IllegalStateException("bad seq value");
+					}
+					return Optional.of(resultSet.getLong(1));
+				}
+				return Optional.empty();
+			}
+		}
+	}
+
+	@Override
+	protected boolean createMissingSeqEntry(Connection connection, String name, String partition, long nextValue)
+			throws SQLException {
+		final Timestamp now = Timestamp.valueOf(LocalDateTime.now());
+		final String sql = POSTGRESQL_INSERT_IGNORE.replace("$TABLE_NAME", tableName);
+		log.debug(sql);
+		try (PreparedStatement statement = connection.prepareStatement(sql)) {
+			statement.setString(1, name);
+			statement.setString(2, partition);
+			statement.setLong(3, nextValue);
+			statement.setTimestamp(4, now);
+			log.debug(String.format("param: [%s] [%s] [%d] [%s]", name, partition, nextValue, now));
+			int rows = statement.executeUpdate();
+			log.debug(String.format("update rows: %d", rows));
+			return rows > 0;
+		}
+	}
+
+	@Override
+	protected boolean updateSeqValue(Connection connection, String name, String partition, long nextValueOld,
+			long nextValueNew) throws SQLException {
+		final Timestamp now = Timestamp.valueOf(LocalDateTime.now());
+		final String sql = POSTGRESQL_UPDATE_VALUE.replace("$TABLE_NAME", tableName);
+		log.debug(sql);
+		try (PreparedStatement statement = connection.prepareStatement(sql)) {
+			statement.setLong(1, nextValueNew);
+			statement.setTimestamp(2, now);
+			statement.setString(3, name);
+			statement.setString(4, partition);
+			statement.setLong(5, nextValueOld);
+			log.debug(String.format("param: [%d] [%s] [%s] [%s] [%d]", nextValueNew, now.toString(), name, partition,
+					nextValueOld));
+			int rows = statement.executeUpdate();
+			log.debug(String.format("update rows: %d", rows));
+			return rows > 0;
+		}
+	}
+
+	@Override
+	protected Connection getConnection() {
+		try {
+			return dataSource.getConnection();
+		}
+		catch (Exception e) {
+			log.warn(e.getMessage(), e);
+			throw new SeqException(e.getMessage(), e);
+		}
+	}
+
+	@Override
+	public AddState tryAddAndGet(String name, String partition, int delta, int maxReTry) {
+		try (Connection connection = getConnection()) {
+			return addAndGet(connection, name, partition, delta).map(val -> AddState.success(val - delta, val, 1))
+					.orElseThrow(() -> new SeqException(String.format("Not exist: %s %s", name, partition)));
+		}
+		catch (SQLException e) {
```

Comment:
Codacy found an issue: [These nested if statements could be combined](https://app.codacy.com/gh/power4j/sequence/pullRequest?prid=5928643)